### PR TITLE
Feature/new send flow [PIL-606]

### PR DESCRIPTION
**Pull request by vyvil**
This PR updates sending flow - now all assets are being sent from the same screen and using the same flow (except for synthetics).

Further changes will be needed once we have new address book.
And confirmation UI will be added in separate PR later in order to reduce the size of this one (🐋).

### CODE REVIEW

File: src/components/SelectorOptions/SelectorOptions.js [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+  options?: Option[],
+  optionTabs?: OptionTabs[],
+  optionsTitle?: string,
+  searchPlaceholder?: string,
+  theme: Theme,
+  noImageFallback?: boolean,
+  inputIconName?: string,
+  iconProps?: Object,
+  activeAccountAddress: string,
+  onHidden: () => void,
```
Comment (IAmKio) [inline src/components/SelectorOptions/SelectorOptions.js:68]: Just out of curiosity, why do we have `() => void` in a few places? I've seen this around, are we overriding prop default actions here?

File: src/components/SelectorOptions/SelectorOptions.js [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+const MIN_QUERY_LENGTH = 2;
+
+
+class SelectorOptions extends React.Component<Props, State> {
+  searchInput: TextInput;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: '',
```
Comment (IAmKio) [inline src/components/SelectorOptions/SelectorOptions.js:152]: I know `''` is falsey but i always prefer `null`... just a personal preference. I always worry that some tools or code evaluate `''` to be _something_ as opposed to _nothing_.

File: src/components/SelectorOptions/SelectorOptions.js [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+      customAddressAsAnOption: null,
+      hasSearchError: false,
+      activeTab: this.props.optionTabs ? this.props.optionTabs[0]?.id : '',
```
Comment (IAmKio) [inline src/components/SelectorOptions/SelectorOptions.js:155]: Same as above in regards to `''` but maybe activeTab is referenced by `''` which is valid? Please let me know if that's the case so i understand in future :)

File: src/components/TextInput/TextInput.js [hunk 4]
```diff
@@ -515,14 +371,14 @@ class TextInput extends React.Component<Props, State> {
...
-  }
+  };
 
   optionKeyExtractor = (option) => {
     const { optionKeyExtractor } = this.props;
     if (optionKeyExtractor) {
       return optionKeyExtractor(option);
     }
-    return option.value;
+    return option?.value;
```
Comment (IAmKio) [inline src/components/TextInput/TextInput.js:381]: 👍

File: src/screens/SendSynthetic/SendSyntheticAmount.js [hunk 5]
```diff
@@ -17,236 +17,190 @@
...
+  handleReceiverSelect = (value: Option, onSuccess?: () => void) => {
+    const { navigation } = this.props;
+    const { name, hasSmartWallet } = value;
+
+    if (hasSmartWallet) {
+      this.setReceiver(value, onSuccess);
+    } else {
+      Alert.alert(
+        'This user is not on Pillar Network',
+        'You both should be connected to Pillar Network in order to be able to send instant transactions for free',
```
Comment (IAmKio) [inline src/screens/SendSynthetic/SendSyntheticAmount.js:136]: Remember to suggest what the user could do: "Why not ask this user to install the Pillar Wallet?" for example...

File: src/screens/SendSynthetic/SendSyntheticAmount.js [hunk 6]
```diff
@@ -17,236 +17,190 @@
...
+        [
+          { text: 'Open Chat', onPress: () => navigation.navigate(CHAT, { username: name }) },
```
Comment (IAmKio) [inline src/screens/SendSynthetic/SendSyntheticAmount.js:138]: Just a reminder that this feature is now dead:
https://github.com/pillarwallet/pillarwallet/pull/2095

File: src/screens/SendSynthetic/SendSyntheticAmount.js [hunk 7]
```diff
@@ -17,236 +17,190 @@
...
-      this.setState({ inputHasError: true });
-    } else if (inputHasError) {
-      this.setState({ inputHasError: false });
+  setReceiver = async (value: Option, onSuccess?: () => void) => {
+    const { receiverEnsName, receiver } = await getContactsEnsName(value?.ethAddress);
+    let stateToUpdate = {};
+    if (!receiver) {
+      Toast.show({
+        title: 'Invalid ENS Name',
+        message: 'Could not get address',
```
Comment (IAmKio) [inline src/screens/SendSynthetic/SendSyntheticAmount.js:153]: Let's be a bit nicer here... this is our grandma after all! 👵 

```suggestion
        title: 'Sorry, we could not find that ENS name!',
        message: 'Could you please check and try again?',
```

File: src/selectors/contacts.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import { createSelector } from 'reselect';
+import { isCaseInsensitiveMatch } from 'utils/common';
+import { contactsSelector, contactsSmartWalletAddressesSelector } from './selectors';
+
+export const contactsForSendFlowSelector = createSelector(contactsSelector, (contacts) => {
```
Comment (IAmKio) [inline src/selectors/contacts.js:25]: Is this anything to do with connections?
https://github.com/pillarwallet/pillarwallet/pull/2095

File: src/utils/contacts.js [hunk 9]
```diff
@@ -20,14 +20,14 @@
...
 import { addressesEqual } from './assets';
-import { isCaseInsensitiveMatch } from './common';
+import { isCaseInsensitiveMatch, resolveEnsName } from './common';
+import { isEnsName } from './validators';
 
 export const getUserName = (contact: ?Object) => {
-  if (!contact || !contact.username) {
-    return '';
-  }
+  if (!contact || !contact.username) return '';
```
Comment (IAmKio) [inline src/utils/contacts.js:30]: I'm starting to wonder, when you specify `username` are we actually talking about the *ENS* name?

File: src/utils/contacts.js [hunk 10]
```diff
@@ -60,3 +60,28 @@ export const findContactIdByUsername = (contacts: ApiUser[], username: string):
...
+  let receiverEnsName = '';
+  let receiver = '';
+  if (!address) return Promise.resolve({ receiverEnsName, receiver });
+
+  if (isEnsName(address)) {
+    const resolvedAddress = await resolveEnsName(address);
+    if (!resolvedAddress) {
+      Toast.show({
+        title: 'Invalid ENS name',
+        message: 'Could not get address',
```
Comment (IAmKio) [inline src/utils/contacts.js:74]: Same as above 👵

File: src/actions/walletConnectActions.js [hunk 11]
```diff
@@ -596,7 +596,7 @@ export const initWalletConnectSessions = () => {
 
     if (connectors.length !== initialConnectors.length) {
       Toast.show({
-        type: 'error',
+        type: 'warning',
         title: 'Session error',
         message: 'Some WalletConnect sessions could not be started',
```
[the remark this answers is not in the document]
Comment (jdnmr) [reply src/actions/walletConnectActions.js:601]: @vyvil You can just add "Please try again." to be consistent with #2101

File: .flowconfig [hunk 12]
```diff
@@ -112,6 +112,10 @@ module.name_mapper='^abi' -> '<PROJECT_ROOT>/src/abi'
 module.name_mapper='^selectors\/\(.*\)$' -> '<PROJECT_ROOT>/src/selectors/\1'
 module.name_mapper='^selectors' -> '<PROJECT_ROOT>/src/selectors'
 
+module.name_mapper='^containers\/\(.*\)$' -> '<PROJECT_ROOT>/src/containers/\1'
+module.name_mapper='^containers' -> '<PROJECT_ROOT>/src/containers'
+
+
```
Comment (poocart) [inline .flowconfig:118]:
```suggestion
```

File: src/components/Selector/Selector.js [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+  searchPlaceholder?: string,
+  horizontalOptionsData?: HorizontalOption[],
+  wrapperStyle?: Object,
+  noOptionImageFallback?: boolean,
+  hasQRScanner?: boolean,
+  disableSelfSelect?: boolean,
+  activeAccountAddress?: string,
+  allowEnteringCustomAddress?: boolean,
+};
+type State = {
```
Comment (poocart) [inline src/components/Selector/Selector.js:56]:
```suggestion
};

type State = {
```

File: src/components/EventDetails/EventDetails.js [hunk 14]
```diff
@@ -640,11 +640,15 @@ export class EventDetail extends React.Component<Props, State> {
...
 
-  sendSynthetic = async () => {
-    const { onClose, navigation } = this.props;
+  sendSynthetic = async (_contact: ?ApiUser, relatedAddress: string) => {
+    const { onClose, navigation, ensRegistry } = this.props;
     onClose();
     await this.switchToSW();
-    navigation.navigate(SEND_SYNTHETIC_ASSET);
+
+    const contactFromAddress = relatedAddress
```
[the remark this answers is not in the document]
Comment (vyvil) [reply src/components/EventDetails/EventDetails.js:606]: I've removed the use of contacts. Here I'm creating a "contact-look-alike" from address to be used in send flow.
I could rename it to receiver, but we will have contacts book later on so we would still have contacts. 
What do you think? :)

File: src/components/SelectorOptions/SelectorOptions.js (hunk 1 above)
Comment (vyvil) [reply src/components/SelectorOptions/SelectorOptions.js:68]: This is just a way to type a function for linting :)

File: src/components/SelectorOptions/SelectorOptions.js (hunk 3 above)
Comment (vyvil) [reply src/components/SelectorOptions/SelectorOptions.js:155]: Nope, I've put `'' `as `null` value here. Will change it to null :)

File: src/selectors/contacts.js (hunk 8 above)
Comment (vyvil) [reply src/selectors/contacts.js]: It does, I've removed it. Thanks!
Wasn't aware that we are not allowing to send tokens to existing contacts ;/

File: src/utils/contacts.js (hunk 9 above)
Comment (vyvil) [reply src/utils/contacts.js:26]: Nope, here username is username of the (now legacy) contact. I'm removing it as no longer necessary.

File: src/screens/SendSynthetic/SendSyntheticAmount.js (hunk 7 above)
Comment (IAmKio) [reply src/screens/SendSynthetic/SendSyntheticAmount.js]: Thank you - it's the little things in life 😄

File: src/components/SelectorOptions/SelectorOptions.js (hunk 1 above)
Comment (IAmKio) [reply src/components/SelectorOptions/SelectorOptions.js:68]: Thanks for the explanation! 😃

File: src/components/EventDetails/EventDetails.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (IAmKio) [reply src/components/EventDetails/EventDetails.js:606]: Let's leave it for now :) especially given that the local address book is coming soon.

**Pull request merged**